### ISSUE TITLE
`typing.cast` is ignored in case of `no-member` error checker

### ISSUE THREAD

**Issue by PiotrBatko-BlockchainWares** [PyCQA/pylint#6784]
### Bug description

In following code Pylint reports `no-member` error. It's right, because `list_wrapper` hasn't such member. I used `typing.cast` and type annotations to force Pylint to think, that `list_wrapper` has different type, but Pylint ignores it.
```
# pylint: disable=missing-module-docstring, missing-class-docstring, too-few-public-methods, missing-function-docstring

import typing

class ListWrapper:
    pass

def some_func():
    list_wrapper: typing.List = typing.cast(typing.List, ListWrapper())
    list_wrapper.append(42)
```

Of course, this simplified example has no sense and such Python code will always fail at runtime. I have a little bit more complicated real use case, with `__setattr__` and `__getattr__` methods, where I assign to member variable such wrapper, like in example above, but overridden `__getattr__` will return different type, which will not cause error at runtime. I wanted to use `typing.cast` to inform Pylint, that returned types for member variables will be different, but it has no effect.

I am not sure if it is a bug or I am using Pylint incorrectly. I can prepare simplified example of my real use case if it would be helpful.

### Command used

```shell
python -m pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:12:4: E1101: Instance of 'ListWrapper' has no 'append' member (no-member)

------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 1.67/10, +0.00)
```

### Expected behavior

No error reported.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.8.13 (default, Apr 19 2022, 00:53:22) 
[GCC 7.5.0]
```

### OS / Environment

Ubuntu 20.04.4 LTS

**Comment by jacobtylerwalls** [comment]
Duplicate of #4534, which was more or less wontfixed in the discussion at https://github.com/PyCQA/astroid/pull/1076#issuecomment-883583488 unless we can find a way to deal with subtle `TypeVar` issues.

(Daniel's example does not raise any `no-member`. ✔️ )
